### DISCUSSION TITLE
How to disable dns resolve action?

### DISCUSSION THREAD

**Discussion by huang-x-h** [apache/apisix#7119] [Q&A]
Backgroud: we delopy the apisix on **a specail network, it can not access dns server and website directly**. 

So we set the `HTTPS_PROXY` to bridge, but when we config the upstream for a uri, such as `https://restapi.amap.com`, the apisix will parse the domain name, and i will be error like this

![fa5ec0bce08c2a11544f84aa94a4e3b](https://user-images.githubusercontent.com/381860/170023728-fb2007f7-9b49-4fd2-bc4f-005d72743edc.png)

Can we disable the dns resolve? so it can use `HTTPS_PROXY` to access the website, or another way to fixed this?

**Comment by tokers** [comment]
Try to set the upstream node to the HTTPS proxy address, and set the Host to the original domain.

**Reply by tzssangglass** [reply]
I remember that disabling DNS is not doable? The domain name must be resolved to ip when setting upstream.

**Reply by huang-x-h** [reply]
Yes, we set `HTTPS_PROXY`, use `curl -i https://restapi.amap.com` it's ok, but we use ping `restapi.amap.com` it's broken.

![image](https://user-images.githubusercontent.com/381860/170173852-5e497e49-b505-4643-81ac-1a65aa42aca6.png)

![1653449300(1)](https://user-images.githubusercontent.com/381860/170173612-21bcc5df-0594-40a7-b721-2f6a4b2a5305.png)

Now we want disable upstream dns resolve. @tzssangglass 

> Try to set the upstream node to the HTTPS proxy address, and set the Host to the original domain.

Use this, we should set a nginx to proxy, new modules have been import ... @tokers

**Reply by tzssangglass** [reply]
I don't know much about `HTTPS_PROXY`

**Reply by tokers** [reply]
I think for a while, it's not feasible. As HTTPS Proxy needs clients to send a CONNECT request. This is not supported by APISIX.
